hackathon_site/dashboard/frontend/src/components/general/ProductOverview/ProductOverview.js

```diff
@@ -0,0 +1,196 @@
+import React from "react";
+import Typography from "@material-ui/core/Typography";
+import Link from "@material-ui/core/Link";
+import Button from "@material-ui/core/Button";
+import MenuItem from "@material-ui/core/MenuItem";
+import FormControl from "@material-ui/core/FormControl";
+import Select from "@material-ui/core/Select";
+import LaunchIcon from "@material-ui/icons/Launch";
+import InputLabel from "@material-ui/core/InputLabel";
+import Chip from "@material-ui/core/Chip";
+import styles from "./ProductOverview.module.scss";
+import Alert from "@material-ui/lab/Alert";
+
+import { Formik } from "formik";
+
+const createQuantityList = (number) => {
+    let entry = [];
+
+    for (let i = 1; i <= number; i++) {
+        let valueInString = i + "";
+        entry.push(
+            <MenuItem key={i} role="quantity" value={valueInString}>
+                {i}
+            </MenuItem>
+        );
+    }
+
+    return entry;
+};
+
+export const CartForm = ({
+    availableQuantity,
+    handleSubmit,
+    handleChange,
+    requestFailure,
+    values: { quantity },
+}) => {
+    if (!handleSubmit) {
+        return false;
+    }
+
+    return (
+        handleSubmit && (
+            <>
+                {requestFailure && (
+                    <Alert className={styles.alert} variant="filled" severity="error">
+                        {requestFailure.message}
+                    </Alert>
+                )}
+                <form className={styles.form} onSubmit={handleSubmit}>
+                    <FormControl variant="outlined" className={styles.formControl}>
+                        <InputLabel>Qty</InputLabel>
+                        <Select
+                            value={quantity}
+                            role="selecter"
+                            onChange={handleChange}
+                            label="Qty"
+                            name="quantity"
+                        >
+                            {createQuantityList(availableQuantity)}
+                        </Select>
+                    </FormControl>
+                    <Button
+                        variant="contained"
+                        className={styles.cartButton}
+                        type="submit"
+                        onClick={handleSubmit}
+                        disableElevation
+                    >
+                        ADD TO CART
+                    </Button>
+                </form>
+            </>
+        )
+    );
+};
+
+export const EnhancedCartForm = ({ handleCart, requestFailure, availableQuantity }) => {
```

Comment:
`handleCart` is sort of a weird prop name. Something like `handleSubmit` is fine, and we can pass in the right function on an outer layer.